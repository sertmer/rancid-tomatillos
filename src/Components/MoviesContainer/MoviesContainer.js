import React from 'react';
import './MoviesContainer.scss';
import Movie from '../Movie/Movie';
import { getMovies } from '../../actions/actions';
import { connect } from 'react-redux';

const MoviesContainer = (props) => {
  const { movies } = props;
  if (!movies.length){
    return <h1>Loading...</h1>
  }
    const displayMovies = movies.map(movie => {
      return(
        <Movie
        {...movie}
        key={movie.id}
        />
      )
    })

  return(
    <>
    {displayMovies}
    </>
  )
}

const mapState = state => ({
  movies: state.movies
});

export default connect(mapState)(MoviesContainer);