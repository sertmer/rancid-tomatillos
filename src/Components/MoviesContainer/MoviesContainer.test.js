import React from 'react';
import { MoviesContainer, mapState } from './MoviesContainer';
import { shallow } from 'enzyme';

describe('MoviesContainer', () => {
  it('should match the snapshot when there are movies in state', () => {
    let mockProps = {
      movies: [{
        id: 1,
        title: 'Erties Dirties',
        average_rating: 2,
        poster_path: 'www.weewoo.com'
      }]
    }
    let wrapper = shallow(<MoviesContainer
      movies={mockProps.movies}
    />)

    expect(wrapper.debug()).toMatchSnapshot()
  })

  describe('mapState', () => {
    it('should return an object with the movies data', () => {
      const mockState = {
        movies: [{
          id: 1,
          title: 'Erties Dirties',
          average_rating: 2,
          poster_path: 'www.weewoo.com'
        }]
      }

      const expected = {
        movies: [{
          id: 1,
          title: 'Erties Dirties',
          average_rating: 2,
          poster_path: 'www.weewoo.com'
        }]
      }

      const mappedProps = mapState(mockState);
      expect(mappedProps).toEqual(expected);
    })
  })
})
