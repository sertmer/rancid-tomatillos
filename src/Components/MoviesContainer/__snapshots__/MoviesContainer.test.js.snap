// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match the snapshot when there are movies in state 1`] = `
"<Fragment>
  <Connect(Movie) movie={{...}} />
</Fragment>"
`;
